sympy/solvers/ode.py

```diff
@@ -3392,8 +3388,8 @@ def _ode_factorable_match(eq, func, x0):
     r = None
     if isinstance(eqs, Pow):
         # if f(x)**p=0 then f(x)=0 (p>0)
-        if (expr.exp).is_positive:
```

Comment:
This is obviously not tested since `expr` is not defined- can you add a test?

Reply:
I've just added a test for this. Currently it isn't possible to hit this code because of the way the equation is preprocessed before it reaches this point. At some point I would like to refactor all of that though so that the original ODE is used here so it is worth ensuring that this branch is correct.